core/schema/container.go

```diff
@@ -1317,7 +1319,8 @@ func (s *containerSchema) withMountedTemp(ctx context.Context, parent *core.Cont
 		return nil, err
 	}
 
-	return parent.WithMountedTemp(ctx, path)
+	// TODO: is the optional safe to dereference?
+	return parent.WithMountedTemp(ctx, path, args.Size.Value.Int())
```

Comment:
I believe not checking it means that when it wasn't set it will default to the zero value, which in the case of an int is literally zero; so provided that `WithMountedTemp` understands `0` as "unset" I think this should be okay. It looks like LLB also treats 0 as unset so think this works out 👍

Just for future reference, if you do want to check if the optional was set you can use `.Valid` before calling `.Value`, e.g. as done here https://github.com/sipsma/dagger/blob/9cf2e8bfdb443223bae84329cd157b3ea7b7e061/core/schema/container.go#L639-L644

Reply:
I had noticed the Valid field in my autocomplete, it's what made me worry I was being unsafe by not checking it :)